cf/data/data.py

```diff
@@ -9376,16 +9400,26 @@ def stats(
             if all or locals()[stat]:
                 func = getattr(self, stat)
                 if stat in no_weights:
-                    value = func(squeeze=True)
+                    value = delayed(func)(squeeze=True)
                 else:
-                    value = func(squeeze=True, weights=weights)
+                    value = delayed(func)(squeeze=True, weights=weights)
 
                 out[stat] = value
 
         if all or sample_size:
-            out["sample_size"] = int(self.sample_size())
+            out["sample_size"] = delayed(lambda: self.sample_size())()
 
-        return out
+        if compute:
```

Comment:
Hi Sadie, I think we should return the actual `Data` objects when `compute=False`, rather than `dask.delayed` objects. The former are still lazy, but much more user-friendly.

Reply:
Oh, I see! I misunderstood what you were implying, sorry. Fair enough, that makes sense. I'll update this shortly to the above.